staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -569,3 +571,98 @@ type CSINodeList struct {
 	// items is the list of CSINode
 	Items []CSINode `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.21
+
+// CSIStorageCapacity stores the result of one CSI GetCapacity call.
+// For a given StorageClass, this describes the available capacity in a
+// particular topology segment.  This can be used when considering where to
+// instantiate new PersistentVolumes.
+//
+// For example this can express things like:
+// - StorageClass "standard" has "1234 GiB" available in "topology.kubernetes.io/zone=us-east1"
+// - StorageClass "localssd" has "10 GiB" available in "kubernetes.io/hostname=knode-abc123"
+//
+// The following three cases all imply that no capacity is available for
+// a certain combination:
+// - no object exists with suitable topology and storage class name
+// - such an object exists, but the capacity is unset
+// - such an object exists, but the capacity is zero
+//
+// The producer of these objects can decide which approach is more suitable.
+type CSIStorageCapacity struct {
```

Comment:
Can we document feature gate requirements here too?

Reply:
This object kind is available regardless of the feature gate. This is the key difference to other, earlier alpha and beta object kinds which were only registered when their flag was set.

What we can say here is "This information is consumed by the kube-scheduler if the CSIStorageCapacity beta feature gate is enabled and a CSI driver opts into capacity-aware scheduling with CSIDriver.StorageCapacity.".

Is that okay?

Reply:
As discussed on chat, I added that sentence.